### ISSUE TITLE
BasisTranslator takes 500ms when there's no basis change during optimization loop

### ISSUE THREAD

**Issue by kdk** [Qiskit/qiskit-terra#8637]
### Environment

- **Qiskit Terra version**: main @ 14a0aaf8
- **Python version**: 3.7
- **Operating system**: osx. 11.6

### What is happening?

Transpiling larger circuits at optimization levels >= 1 has a ~500ms per optimization loop overhead from the `BasisTranslator`, even when the circuit is already in the target basis. It seems like there's an early exit condition we're might be missing somewhere. (I had thought there was a similar PR sometime in the past, but I'm failing to find it now.)

### How can we reproduce the issue?

```
>>> from qiskit.circuit.random import random_circuit
>>> from qiskit.providers.fake_provider import *

>>> c = random_circuit(25, 250)
>>> be = FakeSydney()
>>> be.configuration().num_qubits

>>> import logging 
>>> logging.basicConfig(level='INFO')
>>> qk.transpile(c, be)
INFO:qiskit.transpiler.runningpassmanager:Pass: UnitarySynthesis - 2.89583 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Unroll3qOrMore - 1232.69391 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: SetLayout - 0.00739 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: TrivialLayout - 0.16689 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Layout2qDistance - 31.98695 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: VF2Layout - 45.61305 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: DenseLayout - 1.01829 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: FullAncillaAllocation - 0.07987 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: EnlargeWithAncilla - 0.04792 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: ApplyLayout - 222.33415 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: CheckMap - 7.93600 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: BarrierBeforeFinalMeasurements - 12.19201 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: StochasticSwap - 5154.93584 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: VF2PostLayout - 83.45890 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: UnitarySynthesis - 20.30635 (ms)
...snip...
INFO:qiskit.transpiler.runningpassmanager:Pass: CXCancellation - 213.94467 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: UnitarySynthesis - 72.93582 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: UnrollCustomDefinitions - 205.13606 (ms)
INFO:qiskit.transpiler.passes.basis.basis_translator:Begin BasisTranslator from source basis {('sx', 1), ('rz', 1), ('cx', 2), ('x', 1)} to target basis {'sx', 'snapshot', 'x', 'barrier', 'rz', 'delay', 'reset', 'cx', 'id', 'measure'}.
INFO:qiskit.transpiler.passes.basis.basis_translator:Basis translation path search completed in 0.000s.
INFO:qiskit.transpiler.passes.basis.basis_translator:Basis translation paths composed in 0.137s.
INFO:qiskit.transpiler.passes.basis.basis_translator:Basis translation instructions replaced in 0.226s.
INFO:qiskit.transpiler.runningpassmanager:Pass: BasisTranslator - 580.91593 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Depth - 23.11802 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: FixedPoint - 0.01526 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Size - 0.01168 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: FixedPoint - 0.03505 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Optimize1qGatesDecomposition - 1395.60294 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: CXCancellation - 726.86100 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: UnitarySynthesis - 63.44485 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: UnrollCustomDefinitions - 203.38678 (ms)
***
INFO:qiskit.transpiler.passes.basis.basis_translator:Begin BasisTranslator from source basis {('sx', 1), ('rz', 1), ('cx', 2), ('x', 1)} to target basis {'sx', 'snapshot', 'x', 'barrier', 'rz', 'delay', 'reset', 'cx', 'id', 'measure'}.
***
INFO:qiskit.transpiler.passes.basis.basis_translator:Basis translation path search completed in 0.000s.
INFO:qiskit.transpiler.passes.basis.basis_translator:Basis translation paths composed in 0.128s.
INFO:qiskit.transpiler.passes.basis.basis_translator:Basis translation instructions replaced in 0.226s.
***
INFO:qiskit.transpiler.runningpassmanager:Pass: BasisTranslator - 543.08915 (ms)
***
INFO:qiskit.transpiler.runningpassmanager:Pass: Depth - 30.11298 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: FixedPoint - 0.03886 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Size - 0.01931 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: FixedPoint - 0.02694 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: ContainsInstruction - 0.01502 (ms)
INFO:qiskit.compiler.transpiler:Total Transpile Time - 21524.34373 (ms)
```

### What should happen?

We should investigate if the runtime in `BasisTranslator` can be reduced, since it seems like this can be a ~constant time check (in the size of the circuit) if the `BasisTranslator` doesn't have any changes to make.

**Comment by jakelishman** [comment]
There are 3q unitaries in your circuit, so the BasisTranslator is definitely having to do some work there.

**Comment by jakelishman** [comment]
To be less terse: I threw a breakpoint in the middle of the translator, and by the time we were hitting it (after `Unroll3q` and `UnrollCustomDefinitions`), the source basis the translator needed to modify is:
```python
{'cx', 'crz', 'h', 'tdg', 'sdg', 'u3', 'id', 'rzz', 'u2', 'rx', 'cu3', 'y', 'z', 'x', 'ry', 'cu1', 'cy', 'u1', 't', 'ch', 'swap', 'rz', 'cz', 's'}
```
(compared to a target basis `{'cx', 'x', 'sx', 'rz'}`)

**Comment by mtreinish** [comment]
I think https://github.com/Qiskit/qiskit-terra/pull/6940 is the PR you were thinking of. I can take a look at reviving it since it should be an `O(1)`  lookup to check if the dag is in the global basis list case. For `Target` based compilation it will be `O(n)` (where n is number of instructions) since it needs to check that every instruction's qargs are available on the device.

**Comment by mtreinish** [comment]
Besides #6940 the other thought we probably want to update the `EquivalenceLibrary` to build the equivalence graph internally instead of on demand in the pass. This is low hanging fruit and should reduce the overhead of the basis translator when running it without any transformations being needed.

**Comment by mtreinish** [comment]
I reran your recreate script with #6940 applied and the output log was:

```
INFO:qiskit.transpiler.runningpassmanager:Pass: UnitarySynthesis - 1.19925 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: HighLevelSynthesis - 1.11747 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Unroll3qOrMore - 724.56908 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: SetLayout - 0.00668 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: TrivialLayout - 0.08416 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Layout2qDistance - 11.07979 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: VF2Layout - 20.71810 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: DenseLayout - 7.46584 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: FullAncillaAllocation - 0.07153 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: EnlargeWithAncilla - 0.03505 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: ApplyLayout - 113.59024 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: CheckMap - 4.62914 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: BarrierBeforeFinalMeasurements - 6.78396 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: StochasticSwap - 13823.28033 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: VF2PostLayout - 50.66919 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: UnitarySynthesis - 13.52715 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: HighLevelSynthesis - 11.52611 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: UnrollCustomDefinitions - 72.78132 (ms)
INFO:qiskit.transpiler.passes.basis.basis_translator:Begin BasisTranslator from source basis {('u1', 1), ('crz', 2), ('cu1', 2), ('cy', 2), ('swap', 2), ('cu3', 2), ('cx', 2), ('rx', 1), ('ch', 2), ('ry', 1), ('y', 1), ('x', 1), ('rzz', 2), ('tdg', 1), ('sdg', 1), ('cz', 2), ('s', 1), ('id', 1), ('rz', 1), ('h', 1), ('u2', 1), ('z', 1), ('t', 1), ('u3', 1)} to target basis {'x', 'measure', 'id', 'cx', 'rz', 'barrier', 'reset', 'snapshot', 'delay', 'sx'}.
INFO:qiskit.transpiler.passes.basis.basis_translator:Basis translation path search completed in 0.001s.
INFO:qiskit.transpiler.passes.basis.basis_translator:Basis translation paths composed in 0.037s.
INFO:qiskit.transpiler.passes.basis.basis_translator:Basis translation instructions replaced in 1.138s.
INFO:qiskit.transpiler.runningpassmanager:Pass: BasisTranslator - 1212.14056 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: RemoveResetInZeroState - 20.12753 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Depth - 10.55455 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: FixedPoint - 0.01407 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Size - 0.00906 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: FixedPoint - 0.00334 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Optimize1qGatesDecomposition - 1008.13007 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: CXCancellation - 129.81415 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: GatesInBasis - 0.01717 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Depth - 10.36310 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: FixedPoint - 0.01168 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Size - 0.00668 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: FixedPoint - 0.00286 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Optimize1qGatesDecomposition - 697.63064 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: CXCancellation - 269.80209 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: GatesInBasis - 0.01788 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Depth - 11.12223 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: FixedPoint - 0.01192 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Size - 0.00858 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: FixedPoint - 0.00310 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Optimize1qGatesDecomposition - 684.84616 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: CXCancellation - 120.53752 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: GatesInBasis - 0.01645 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Depth - 10.89025 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: FixedPoint - 0.00978 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Size - 0.00691 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: FixedPoint - 0.00310 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Optimize1qGatesDecomposition - 688.08556 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: CXCancellation - 122.23935 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: GatesInBasis - 0.01717 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Depth - 11.51919 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: FixedPoint - 0.01121 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Size - 0.00763 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: FixedPoint - 0.00286 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Optimize1qGatesDecomposition - 827.26979 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: CXCancellation - 119.83681 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: GatesInBasis - 0.01764 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Depth - 10.70118 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: FixedPoint - 0.01001 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Size - 0.00691 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: FixedPoint - 0.00310 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Optimize1qGatesDecomposition - 706.85720 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: CXCancellation - 120.51153 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: GatesInBasis - 0.01502 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Depth - 10.53095 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: FixedPoint - 0.00978 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Size - 0.00715 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: FixedPoint - 0.00310 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Optimize1qGatesDecomposition - 683.69198 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: CXCancellation - 257.98750 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: GatesInBasis - 0.01740 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Depth - 10.22387 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: FixedPoint - 0.01073 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Size - 0.00763 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: FixedPoint - 0.00310 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: ContainsInstruction - 0.00429 (ms)
INFO:qiskit.compiler.transpiler:Total Transpile Time - 24212.71443 (ms)
```

The `GatesInBasis` pass is taking ~0.02 sec and the basis translator is skipped.

**Comment by mtreinish** [comment]
Switching to the backendV2 `SydneyV2` gates in basis is signficantly slower as expected (because it needs to verify each individual instruction to test the full `Target` is satisfied). But it is still faster than the 500ms for basis translator (although I don't know how my local times for an empty basis translator compare):
```
INFO:qiskit.transpiler.runningpassmanager:Pass: UnitarySynthesis - 1.22786 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: HighLevelSynthesis - 1.13297 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Unroll3qOrMore - 720.29305 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: SetLayout - 0.00668 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: TrivialLayout - 0.09274 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Layout2qDistance - 11.43098 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: VF2Layout - 22.09330 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: DenseLayout - 7.18927 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: FullAncillaAllocation - 0.07057 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: EnlargeWithAncilla - 0.03529 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: ApplyLayout - 116.85586 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: CheckMap - 4.81248 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: BarrierBeforeFinalMeasurements - 7.81751 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: StochasticSwap - 14459.17916 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: VF2PostLayout - 46.27037 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: UnitarySynthesis - 10.71382 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: HighLevelSynthesis - 10.40745 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: UnrollCustomDefinitions - 75.62399 (ms)
INFO:qiskit.transpiler.passes.basis.basis_translator:Begin BasisTranslator from source basis {('x', 1), ('cz', 2), ('ry', 1), ('rz', 1), ('z', 1), ('h', 1), ('sdg', 1), ('cy', 2), ('t', 1), ('s', 1), ('crz', 2), ('rzz', 2), ('tdg', 1), ('cu1', 2), ('y', 1), ('swap', 2), ('cx', 2), ('u2', 1), ('id', 1), ('u3', 1), ('ch', 2), ('cu3', 2), ('u1', 1), ('rx', 1)} to target basis {'delay', 'cx', 'reset', 'snapshot', 'measure', 'rz', 'sx', 'barrier', 'x', 'id'}.
INFO:qiskit.transpiler.passes.basis.basis_translator:Basis translation path search completed in 0.001s.
INFO:qiskit.transpiler.passes.basis.basis_translator:Basis translation paths composed in 0.035s.
INFO:qiskit.transpiler.passes.basis.basis_translator:Basis translation instructions replaced in 1.246s.
INFO:qiskit.transpiler.runningpassmanager:Pass: BasisTranslator - 1348.64378 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: RemoveResetInZeroState - 19.77062 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Depth - 11.06429 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: FixedPoint - 0.01287 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Size - 0.00858 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: FixedPoint - 0.00358 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Optimize1qGatesDecomposition - 857.15723 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: CXCancellation - 128.67999 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: GatesInBasis - 140.32817 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Depth - 11.06668 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: FixedPoint - 0.01144 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Size - 0.00858 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: FixedPoint - 0.00310 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Optimize1qGatesDecomposition - 819.68904 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: CXCancellation - 123.15035 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: GatesInBasis - 141.14356 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Depth - 11.16872 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: FixedPoint - 0.01192 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Size - 0.00882 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: FixedPoint - 0.00310 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Optimize1qGatesDecomposition - 682.48248 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: CXCancellation - 277.81892 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: GatesInBasis - 140.89370 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Depth - 11.37209 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: FixedPoint - 0.01168 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Size - 0.00882 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: FixedPoint - 0.00334 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Optimize1qGatesDecomposition - 682.75070 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: CXCancellation - 125.18811 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: GatesInBasis - 140.59687 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Depth - 11.22594 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: FixedPoint - 0.01359 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Size - 0.01001 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: FixedPoint - 0.00334 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Optimize1qGatesDecomposition - 686.28025 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: CXCancellation - 122.96247 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: GatesInBasis - 140.93113 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Depth - 10.32352 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: FixedPoint - 0.01311 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Size - 0.00858 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: FixedPoint - 0.00310 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Optimize1qGatesDecomposition - 828.14288 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: CXCancellation - 121.28234 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: GatesInBasis - 140.67221 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Depth - 10.95366 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: FixedPoint - 0.01168 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Size - 0.01001 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: FixedPoint - 0.00334 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Optimize1qGatesDecomposition - 683.87032 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: CXCancellation - 121.77706 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: GatesInBasis - 140.54227 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Depth - 10.48017 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: FixedPoint - 0.01025 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: Size - 0.00834 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: FixedPoint - 0.00310 (ms)
INFO:qiskit.transpiler.runningpassmanager:Pass: ContainsInstruction - 0.00429 (ms)
INFO:qiskit.compiler.transpiler:Total Transpile Time - 25757.56764 (ms)
```

**Comment by jakelishman** [comment]
Are we considering this fixed by #6940?

(Also, in my above comment, I see now that I'd just looked at the first pass, not the instances in the optimisation loop, so I was just chatting nonsense.)

**Comment by mtreinish** [comment]
I think we can, but it'd be good to confirm with @kdk that his performance issue is addressed. I still want to try and reduce the overhead of `BasisTranslator` and `EquivalenceLibrary` in cases where there is no transformation to perform but we can do that in more targeted issues.

**Comment by kdk** [comment]
Thanks @mtreinish and @jakelishman , checked that this was resolved by #6940 so I think we're good to close this.
